File: TellDontAsk/AlarmsProcedural/AlarmServices/CompositeAlarmService.cs
using System;

namespace AlarmsProcedural.AlarmServices
{
  public static class CompositeAlarmService
  {
    public static void Trigger(Alarm alarm)
    {
      AlarmService.TriggerAlarm(alarm.Nested1);
      AlarmService.TriggerAlarm(alarm.Nested2);
    }

    public static void Disable(Alarm alarm)
    {
      AlarmService.DisableAlarm(alarm.Nested1);
      AlarmService.DisableAlarm(alarm.Nested2);
    }

    public static void Dump(Alarm alarm)
    {
      Console.WriteLine("{ Both: ");
      AlarmService.Dump(alarm.Nested1);
      AlarmService.Dump(alarm.Nested2);
      Console.WriteLine("} ");
    }
  }
}

File: Java/AlarmsProcedural/src/main/java/AlarmServices/CompositeAlarmService.java
package AlarmServices;

import Main.Alarm;

public class CompositeAlarmService {
  public static void trigger(Alarm alarm) {
    AlarmService.triggerAlarm(alarm.nested1);
    AlarmService.triggerAlarm(alarm.nested2);
  }

  public static void disable(Alarm alarm) {
    AlarmService.disableAlarm(alarm.nested1);
    AlarmService.disableAlarm(alarm.nested2);
  }

  public static void dump(Alarm alarm) {
    System.out.println("{ Both: ");
    AlarmService.dump(alarm.nested1);
    AlarmService.dump(alarm.nested2);
    System.out.println("} ");
  }
}
